### ISSUE TITLE
Consider adding `secure_pubspec_urls` to core

### ISSUE THREAD

**Issue by pq** [dart-lang/lints#59]
[secure_pubspec_urls](https://dart-lang.github.io/linter/lints/secure_pubspec_urls.html) would be a good candidate for `core`.

## Lint Descrption

**DO** Use secure urls in `pubspec.yaml`.

Use `https` instead of `http` or `git:`.

**GOOD:**
```yaml
repository: https://github.com/dart-lang/example
```

**BAD:**
```yaml
repository: http://github.com/dart-lang/example
```

```yaml
git:
  url: git://github.com/dart-lang/example/example.git 
```

**Comment by mit-mit** [comment]
Not sure what the value is -- @jonasfj wdyt?

**Comment by jonasfj** [comment]
Github has been [phasing out `git://` ](https://github.blog/2021-09-01-improving-git-protocol-security-github/).

Honestly, if you are using `http://` URLs that's probably a mistake. Considering that you'll be fetching dependencies over HTTP I think there is value in discouraging this.

There could be corner cases where people want to use `http://localhost`, or `http://` on an intranet, but those users can change their lint settings. In general it's a bad idea to use HTTP dependencies.

**Comment by jonasfj** [comment]
You will miss out on pub points if you don't use HTTPS for all links in `pubspec.yaml`. Having this in linter is good because it will help developers discover this before publishing.

**Comment by devoncarew** [comment]
One consideration here is that we can't ignore lints in pubspec files.

**Comment by devoncarew** [comment]
This has been approved for the `core` set.
